vavr/src/main/java/io/vavr/collection/AbstractMultimap.java

```diff
@@ -579,22 +579,26 @@ public String toString() {
     }
 
     protected <JM extends java.util.Map<K, Collection<V>>> JM toJavaMap(JM javaMap) {
-        final Supplier<Collection<V>> javaContainerSupplier;
-        if (containerType == ContainerType.SEQ) {
-            javaContainerSupplier = java.util.ArrayList::new;
-        } else if (containerType == ContainerType.SET) {
-            javaContainerSupplier = java.util.HashSet::new;
-        } else if (containerType == ContainerType.SORTED_SET) {
-            javaContainerSupplier = java.util.TreeSet::new;
-        } else {
-            throw new IllegalStateException("Unknown ContainerType: " + containerType);
-        }
+        final Supplier<Collection<V>> javaContainerSupplier = getJavaContainerSupplier();
         for (Tuple2<K, V> t : this) {
             javaMap.computeIfAbsent(t._1, k -> javaContainerSupplier.get()).add(t._2);
         }
         return javaMap;
     }
 
+    private Supplier<Collection<V>> getJavaContainerSupplier() {
```

Comment:
Better would be to have this as a parameter in the enum constructor and add a getter - then the 

```
default:
                throw new IllegalStateException("Unknown ContainerType: " + containerType);
```

Can't possibly occur. 

```
enum ContainerType {

        SET(
                (Traversable<?> set, Object elem) -> ((Set<Object>) set).add(elem),
                (Traversable<?> set, Object elem) -> ((Set<Object>) set).remove(elem),
                HashSet::new,
        )

        ContainerType(BiFunction<Traversable<?>, Object, Traversable<?>> add,
                BiFunction<Traversable<?>, Object, Traversable<?>> remove,
                 Supplier<Collection<?>> instantiator) {
            this.add = add;
            this.remove = remove;
            this.instantiator = instantiator;
        }

        public Supplier<Collection<?>> getInstantiator() {
             return instantiator;
        }

Reply:
Interesting, lemme' check it, thank you! :)

Reply:
Cool idea @chb0github, applied it in https://github.com/vavr-io/vavr/pull/2112/commits/b5205538133f0ec732e2161bd5b5e5b38c35fa2b

Reply:
@chb0github nice one!